src/api/eth/proposal.rs

```diff
@@ -38,19 +39,30 @@ impl ProposalReq {
     }
 
     // Get mmr root
-    fn mmr_root(store: &Store, leaf: u64) -> String {
-        let mmr = MMR::<_, MergeHash, _>::new(cmmr::leaf_index_to_mmr_size(leaf), store);
+    fn mmr_root(store: &Store, block: u64) -> String {
+        if block == 0 {
+            return "0x00000000000000000000000000000000".into();
+        }
+        let mmr = MMR::<_, MergeHash, _>::new(cmmr::leaf_index_to_mmr_size(block - 1), store);
         format!("0x{}", H256::hex(&mmr.get_root().unwrap_or_default()))
     }
 
     /// Generate mmr proof
-    fn mmr_proof(&self, store: &Store, member: u64) -> Vec<String> {
-        let mmr = MMR::<_, MergeHash, _>::new(cmmr::leaf_index_to_mmr_size(self.last_leaf), store);
+    fn mmr_proof(&self, store: &Store, mut member: u64) -> Vec<String> {
+        if self.target < 1 {
+            return vec![];
+        }
+
+        if member == self.target {
+            member -= 1;
```

Comment:
Suggest to return Error or return empty array.

Reply:
Reasonable, let me test it.